AppTaskQueue/appscale/taskqueue/distributed_tq.py

```diff
@@ -574,19 +557,23 @@ def __check_and_store_task_names(self, request):
     item = TaskName.get_by_key_name(task_name)
     logger.debug("Task name {0}".format(task_name))
     if item:
-      logger.warning("Task already exists")
-      raise apiproxy_errors.ApplicationError(
-        TaskQueueServiceError.TASK_ALREADY_EXISTS)
-    else:
-      new_name = TaskName(key_name=task_name, state=tq_lib.TASK_STATES.QUEUED,
-        queue=request.queue_name(), app_id=request.app_id())
-      logger.debug("Creating entity {0}".format(str(new_name)))
-      try:
-        db.put(new_name)
-      except datastore_errors.InternalError, internal_error:
-        logger.error(str(internal_error))
+      if item.state == TASK_STATES.QUEUED:
+        logger.warning("Task already exists")
         raise apiproxy_errors.ApplicationError(
-          TaskQueueServiceError.DATASTORE_ERROR)
+          TaskQueueServiceError.TASK_ALREADY_EXISTS)
+      else:
+        raise apiproxy_errors.ApplicationError(
+          TaskQueueServiceError.INVALID_TASK_NAME)
```

Comment:
Does GAE somehow differentiate invalid task name like this case (used name) and invalid name containing some special symbols? Is it easy to add some descriptive message here?

Reply:
I don't take my approval)

Reply:
Good catch. I thought when I tested earlier that GAE used InvalidTaskNameError for tombstoned task names, but when I just tested now, it returned TombstonedTaskError.